org.wso2.codequalitymatrices/src/main/java/com/wso2/code/quality/metrics/ApiUtility.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.wso2.code.quality.metrics;
+
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.log4j.Logger;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * This is a utility class for calling APIs.
+ *
+ * @since 1.0.0
+ */
+public final class ApiUtility {
+    private static final Logger logger = Logger.getLogger(ApiUtility.class);
+
+    // to prevent instantiation
+    private ApiUtility() {
+    }
+
+    /**
+     * This is used for calling the REST APIs.
+     *
+     * @param httpGet Instance of the relevant httpGet
+     * @return String representation of the json response
+     * @throws CodeQualityMetricsException results
+     */
+    public static String callApi(HttpGet httpGet) throws CodeQualityMetricsException {
+        BufferedReader bufferedReader = null;
+        CloseableHttpClient httpClient;
+        CloseableHttpResponse httpResponse = null;
+        httpClient = HttpClients.createDefault();
+        String jsonText;
+        try {
+            httpResponse = httpClient.execute(httpGet);
+            int responseCode = httpResponse.getStatusLine().getStatusCode();
+            if (responseCode == 200) {
+                //success
+                bufferedReader = new BufferedReader(new InputStreamReader(httpResponse.getEntity().getContent(),
```

Comment:
Same with `httpResponse.getEntity()`. When chaining calls, better to verify for possible NPEs.